services/apps/alcs/src/alcs/admin/tag-category/tag-category.controller.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOAuth2 } from '@nestjs/swagger';
+import * as config from 'config';
+import { RolesGuard } from '../../../common/authorization/roles-guard.service';
+import { UserRoles } from '../../../common/authorization/roles.decorator';
+import { AUTH_ROLE } from '../../../common/authorization/roles';
+import { TagCategoryDto } from './tag-category.dto';
+import { TagCategoryService } from './tag-category.service';
+
+@Controller('tag-category')
+@ApiOAuth2(config.get<string[]>('KEYCLOAK.SCOPES'))
+@UseGuards(RolesGuard)
+export class TagCategoryController {
+
+  constructor(private service: TagCategoryService) {}
+
+  @Get('/:pageIndex/:itemsPerPage')
+  @UserRoles(AUTH_ROLE.ADMIN)
+  async fetch(
+    @Param('pageIndex') pageIndex: number,
+    @Param('itemsPerPage') itemsPerPage: number,
+    @Query('search') search?: string,
+  ) {
+    const result = await this.service.fetch(pageIndex, itemsPerPage, search);
+    return { data: result[0], total: result[1] };
+  }
+
+  @Post('')
+  @UserRoles(AUTH_ROLE.ADMIN)
+  async create(@Body() createDto: TagCategoryDto) {
+    return await this.service.create(createDto);
+  }
+
+  @Put('/:uuid')
```

Comment:
It looks like in the downstream service you are updating only one field of the entity which is the name. Since it is a partial update, would it be better to use PATCH request instead of PUT?